Traceback (most recent call last):
  File "/data/pulkitag/models/jyop/code/nanoGPT/train.py", line 268, in <module>
    losses = estimate_loss()
             ^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/code/nanoGPT/train.py", line 228, in estimate_loss
    logits, loss = model(X, Y)
                   ^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/eval_frame.py", line 465, in _fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 1269, in __call__
    return self._torchdynamo_orig_callable(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 1064, in __call__
    result = self._inner_convert(
             ^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 526, in __call__
    return _compile(
           ^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 952, in _compile
    raise InternalTorchDynamoError(
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 924, in _compile
    guarded_code = compile_inner(code, one_graph, hooks, transform)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 666, in compile_inner
    return _compile_inner(code, one_graph, hooks, transform)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_utils_internal.py", line 87, in wrapper_function
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 699, in _compile_inner
    out_code = transform_code_object(code, transform)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/bytecode_transformation.py", line 1322, in transform_code_object
    transformations(instructions, code_options)
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 219, in _fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 634, in transform
    tracer.run()
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 2796, in run
    super().run()
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 983, in run
    while self.step():
          ^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 895, in step
    self.dispatch_table[inst.opcode](self, inst)
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 582, in wrapper
    return inner_fn(self, inst)
           ^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 2279, in CALL
    self._call(inst)
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 2273, in _call
    self.call_function(fn, args, kwargs)
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 830, in call_function
    self.push(fn.call_function(self, args, kwargs))  # type: ignore[arg-type]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/variables/lazy.py", line 156, in realize_and_forward
    return getattr(self.realize(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/variables/nn_module.py", line 899, in call_function
    return variables.UserFunctionVariable(fn, source=source).call_function(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/variables/functions.py", line 324, in call_function
    return super().call_function(tx, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/variables/functions.py", line 111, in call_function
    return tx.inline_user_function_return(self, [*self.self_args(), *args], kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 836, in inline_user_function_return
    return InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 3011, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 3139, in inline_call_
    tracer.run()
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 983, in run
    while self.step():
          ^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 895, in step
    self.dispatch_table[inst.opcode](self, inst)
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 582, in wrapper
    return inner_fn(self, inst)
           ^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 301, in impl
    self.push(fn_var.call_function(self, self.popn(nargs), {}))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/variables/builtin.py", line 967, in call_function
    return handler(tx, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/variables/builtin.py", line 848, in builtin_dispatch
    rv = fn(tx, args, kwargs)
         ^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/variables/builtin.py", line 766, in call_self_handler
    result = self_handler(tx, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/variables/builtin.py", line 1472, in call_getitem
    return args[0].call_method(tx, "__getitem__", args[1:], kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/variables/lists.py", line 437, in call_method
    return super().call_method(tx, name, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/variables/lists.py", line 392, in call_method
    return super().call_method(tx, name, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/variables/lists.py", line 131, in call_method
    return self.getitem_const(tx, value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/pulkitag/models/jyop/miniforge3/envs/coconut/lib/python3.12/site-packages/torch/_dynamo/variables/lists.py", line 107, in getitem_const
    return self.items[index]
           ~~~~~~~~~~^^^^^^^
torch._dynamo.exc.InternalTorchDynamoError: IndexError: list index out of range

from user code:
   File "/data/pulkitag/models/jyop/code/nanoGPT/model.py", line 228, in forward
    x = self.transformer.routers[layer_idx](x, expert_outputs)
  File "/data/pulkitag/models/jyop/code/nanoGPT/model.py", line 129, in forward
    combined_output += expert_outputs[i] * routing_weights[:, :, i:i+1]

Set TORCH_LOGS="+dynamo" and TORCHDYNAMO_VERBOSE=1 for more information


You can suppress this exception and fall back to eager by setting:
    import torch._dynamo
    torch._dynamo.config.suppress_errors = True
